### PR TITLE
:zap: add automatic validating skills

### LINKED ISSUE #73: Automaticaly validate skills when selecting the desire level

### PR DESCRIPTION
**Pull request by MailyLehoux**
resolve #73

### CODE REVIEW

**Comment by ebriand** [comment]
I wonder if the skill is validated without click on the "Add Skill" button, is this button still useful? Maybe we need to remove it? Maybe it's better to ask Stéphane about that 🙂

**Comment by Ked57** [comment]
> 
> 
> I wonder if the skill is validated without click on the "Add Skill" button, is this button still useful? Maybe we need to remove it? Maybe it's better to ask Stéphane about that 🙂

I think it's useful if the user wants to update their skill, it makes more sense to click on update than re-entering all of the values. But it might ok to remove it when the skill is new

File: src/components/AddOrEditSkillModale.tsx [hunk 1]
```diff
@@ -16,12 +16,17 @@ const AddOrEditSkillModale = ({
...
   const [skillLevel, setSkillLevel] = useState(skill?.level || 0);
   const [desireLevel, setDesireLevel] = useState(skill?.desire || 0);
 
   const onAddButtonClick = () => {
     callback({ ...skill, level: skillLevel, desire: desireLevel });
   };
+  useEffect(() => {
+    if (autoValidation)
+      onAddButtonClick();
+  }, [!autoValidation]);
```
Comment (Ked57) [inline src/components/AddOrEditSkillModale.tsx:29]: I think there is a simpler way to do this than adding a new variable. Since clicking on one of the buttons changes the desire level you could just do 
```ts
useEffect(() => {
    if (skillLevel != 0 && desireLevel != 0) {
      onAddButtonClick();
    }
  }, [desireLevel]);
```

File: src/components/AddOrEditSkillModale.tsx [hunk 2]
```diff
@@ -83,7 +88,11 @@ const AddOrEditSkillModale = ({
             {[1, 2, 3, 4, 5].map((index) => (
               <button
                 key={`desire-${index}`}
-                onClick={() => setDesireLevel(index)}
+                onClick={() => {
+                  setDesireLevel(index);
+                  if (skillLevel > 0 && index > 0)
+                    setautoValidation(true)
+                }}
```
Comment (Ked57) [inline src/components/AddOrEditSkillModale.tsx:95]: My previous suggestion would revert that change

File: src/components/AddOrEditSkillModale.tsx [hunk 3]
```diff
@@ -16,12 +16,17 @@ const AddOrEditSkillModale = ({
 }: AddOrEditSkillModaleProps) => {
   const { t } = useContext(i18nContext);
   const [navState, setNavState] = useState("knowledge");
+  const [autoValidation, setautoValidation] = useState(false);
```
Comment (Ked57) [inline src/components/AddOrEditSkillModale.tsx:19]: and this one

**Review by Ked57 — COMMENTED** [review]
Great work nonetheless ! Don't forget to run `npm run prettier:write` before pushing code so the CI doesn't fail
If you want we can setup a meet to discuss this further 😃

**Review by Ked57 — CHANGES_REQUESTED** [review]
I would add brackets on this if statement to make it more readable

File: src/components/AddOrEditSkillModale.tsx [hunk 4]
```diff
@@ -83,7 +88,11 @@ const AddOrEditSkillModale = ({
             {[1, 2, 3, 4, 5].map((index) => (
               <button
                 key={`desire-${index}`}
-                onClick={() => setDesireLevel(index)}
+                onClick={() => {
+                  setDesireLevel(index);
+                  if (skillLevel > 0 && index > 0)
+                    setautoValidation(true)
```
Comment (hgwood) [inline src/components/AddOrEditSkillModale.tsx:94]: Can't `onAddButtonClick` be directly called here?

Comment (Ked57) [reply src/components/AddOrEditSkillModale.tsx:94]: `setDesireLevel` is not synchronous, `onAddButtonClick` would use the old desire level

**Review by Ked57 — COMMENTED** [review]
I though of another way to do this without the autoValidation variable

```ts
 useEffect(() => {
    if (
      skillLevel === 0 ||
      desireLevel === skill?.desire ||
      desireLevel === 0
    ) {
      return;
    }
    onAddButtonClick();
  }, [desireLevel]);
```
This would only autovalidate in cases where
- skill is new, both desire and skill have been set
- skill isn't new, at least the desire has been updated

File: src/components/AddOrEditSkillModale.tsx (hunk 4 above)
Comment (hgwood) [reply src/components/AddOrEditSkillModale.tsx:94]: Right!

File: src/components/AddOrEditSkillModale.tsx [hunk 5]
```diff
@@ -83,7 +93,9 @@ const AddOrEditSkillModale = ({
             {[1, 2, 3, 4, 5].map((index) => (
               <button
                 key={`desire-${index}`}
-                onClick={() => setDesireLevel(index)}
+                onClick={() => {
+                  setDesireLevel(index);
+                }}
```
Comment (Ked57) [inline src/components/AddOrEditSkillModale.tsx:98]: This is causing a merge conflict for some reason, I think we should revert this change since it doesn't bring anything.
Then we can merge 😃

Comment (MailyLehoux) [reply src/components/AddOrEditSkillModale.tsx]: done 👍

**Review by Ked57 — APPROVED** [review]
